adafruit_pm25/uart.py

```diff
@@ -46,38 +51,207 @@
 from digitalio import Direction
 from . import PM25
 
+PLANTOWER_HEADER = b"\x42\x4D"
+
+PLANTOWER_CMD_MODE_PASSIVE = b"\xE1\x00\x00"
+PLANTOWER_CMD_MODE_ACTIVE = b"\xE1\x00\x01"
+PLANTOWER_CMD_READ = b"\xE2\x00\x00"
+PLANTOWER_CMD_SLEEP = b"\xE4\x00\x00"
+PLANTOWER_CMD_WAKEUP = b"\xE4\x00\x01"
+
+UART_RETRY_COUNT = 5
+FRAME_RETRY_COUNT = 3
+MAX_FRAME_SIZE = 32
+
 
 class PM25_UART(PM25):
     """
     A driver for the PM2.5 Air quality sensor over UART
     """
 
-    def __init__(self, uart, reset_pin=None):
+    def __init__(self, uart, reset_pin=None, set_pin=None, mode="passive"):
```

Comment:
The keyword argument of `mode="passive"` is going to change the behaviour for existing users of this library. It seems undesirable for them to get this change without requesting it explicitly by setting `mode` from the program?

Reply:
That's a good point. I spent a bit of time mulling over it and looking for guidance from docs before settling on this. Here's my general train of thought.

At a high level, behabior should be unchanged. The end result should be identical operation, based on the driver's design. When a request is made for a `read()` the user will get a reading as expected.

The only difference for a user would be at a lower level beyond the driver's abstraction. Rather than having a stream of frames coming in every 900-1000ms over the UART, the UART would only see incoming data frames in response to requests (triggered by `read()`.

With this constructor arg set up the user could request an "active" mode configuration if their use case required it. For example, if their implmentation was reading the UART frames outside of the CircuitPython program/MCU, or using lower-level UART reads outside of our driver.

Additionally, I noticed that active-mode reads are generally unreliable, leading to errors from if frames are picked up in the middle and/or truncated before the checksum comes in. That was the quirk that led me down this rabbit hole in the first place. 😄 

That was my original intent and internal argument though. I can see the case for making the default "active", retaining that convention for UART devices in this driver is paramount.

Reply:
I didn't address this in my commits this morning, but I'd be happy to make any changes suggested by maintainers here.

Its my first time working on a proper CircuitPython driver so I'm very open to guidance and discussion. :)